tasks/LaserScanTask.cpp

```diff
@@ -71,39 +74,74 @@ void LaserScanTask::cleanupHook()
     LaserScanTaskBase::cleanupHook();
 }
 
-void LaserScanTask::readInput(ConstLaserScanStampedPtr & laserScanMSG)
+void LaserScanTask::readInput(ConstLaserScanStampedPtr& laserScanMSG)
 {
     if (state() != RUNNING) {
         return;
     }
 
     lock_guard<mutex> readGuard(readMutex);
-
+    unsigned int scan_size = laserScanMSG->scan().ranges_size();
     double range_min = laserScanMSG->scan().range_min();
     double range_max = laserScanMSG->scan().range_max();
-
-    scan.time = getCurrentTime(laserScanMSG->time());
-    scan.minRange = meters_to_milimeters(range_min);
-    scan.maxRange = meters_to_milimeters(range_max);
-    scan.angular_resolution = laserScanMSG->scan().angle_step();
-    scan.start_angle = laserScanMSG->scan().angle_min();
-
-    unsigned int scan_size = laserScanMSG->scan().ranges_size();
-    scan.ranges.resize(scan_size);
-
-    for (unsigned int i = 0; i < scan_size; ++i) {
-        double range = laserScanMSG->scan().ranges(i);
-
-        if (range >= range_max) {
-            scan.ranges[i] = base::samples::TOO_FAR;
+    if (laserScanMSG->scan().has_vertical_count()) {
+        if (m_depth_map.vertical_size != laserScanMSG->scan().vertical_count()) {
```

Comment:
That does a very long method that is essentially two methods. Create two methods, outputDepthMap and outputLaserScan, and call them each in one branch of this `if()`